File: src/Learnify/Learnify.Core/Dto/Course/LessonDtos/LessonAddOrUpdateRequest.cs
using Learnify.Core.Dto.Course.Video;
using Learnify.Core.Enums;

namespace Learnify.Core.Dto.Course.LessonDtos;

public class LessonAddOrUpdateRequest
{
    public string Id { get; set; }
    public int ParagraphId { get; set; }
    public string Title { get; set; }
    public string EditedLessonId { get; set; }
    public Language PrimaryLanguage { get; set; }
    public VideoAddOrUpdateRequest Video { get; set; }
}

File: front/client/src/app/Models/Course/Lesson/LessonAddOrUpdateRequest.ts
import { Language } from "../../enums/Language";
import { VideoAddOrUpdateRequest } from "./Video/VideoAddOrUpdateRequest";

export interface LessonAddOrUpdateRequest {
    id: string | null;
    paragraphId: number;
    title: string;
    editedLessonId: string;
    primaryLanguage: Language;
    video: VideoAddOrUpdateRequest;
}